### PR TITLE
Decompile unk_0204F500

### PR DESCRIPTION
**Pull request by cbt6**

### CODE REVIEW

File: src/scrcmd_battle_arcade.c [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+    unk->unk0c = a1;
+    TaskManager_Call(fsys->taskman, sub_02050150, unk);
+}
+
+static BOOL sub_02050150(TaskManager *taskManager) {
+    FieldSystem* fsys = TaskManager_GetFieldSystem(taskManager);
+    UnkStruct_0204FBDC *unk = TaskManager_GetEnv(taskManager);
+    switch (unk->state) {
+        case 0:
+            unk->state = sub_020501B8(unk, fsys, 0xb);
```
Comment (red031000) [inline src/scrcmd_battle_arcade.c:174]: HEAP_ID_FIELD

File: src/scrcmd_battle_castle.c [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+#include "unk_02035900.h"
+#include "unk_0203DFA4.h"
+#include "unk_0203E348.h"
+#include "unk_0205BB1C.h"
+#include "unk_0205BFF0.h"
+#include "unk_02078E30.h"
+#include "unk_02088288.h"
+#include "unk_02091564.h"
+
+typedef enum BattleCastleChallengeType {
```
Comment (red031000) [inline src/scrcmd_battle_castle.c:32]: again, should be in headers, if this is the same as the arcade, possibly merging the two definitions?

File: src/scrcmd_20.c [hunk 3]
```diff
@@ -335,3 +382,151 @@ static u32 sub_0204F4D8(UnkStruct_0204F284 *a0, FieldSystem *fsys) {
...
+    void *unk1 = sub_020312C4(fsys->savedata, 0x20, &unk0);
+    if (unk0 == 1) {
+        for (u32 i = 0; i < 0x1ed; i++) {
+            winStreak += sub_020312E0(fsys->savedata, unk1, 0, i);
+        }
+    }
+    if (unk1 != NULL) {
+        FreeToHeap(unk1);
+    }
+    if (winStreak > 10000) {
```
[the remark this answers is not in the document]
Comment (cbt6) [reply src/scrcmd_20.c:498]: Seems pointless? It'll never be used elsewhere.

File: src/scrcmd_battle_arcade.c [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+#include "unk_02035900.h"
+#include "unk_0203DFA4.h"
+#include "unk_0203E348.h"
+#include "unk_0205BB1C.h"
+#include "unk_0205BFF0.h"
+#include "unk_02078E30.h"
+#include "unk_02088288.h"
+#include "unk_02091564.h"
+
+typedef enum BattleArcadeChallengeType {
```
[the remark this answers is not in the document]
Comment (cbt6) [reply src/scrcmd_battle_arcade.c:32]: Why? These typedefs are only ever used inside their respective source files (same applies to the other typedefs).

**Review by red031000 — APPROVED** [review]

**Pull request merged**